core/modules/editor/operations/text/wrap-selection.js

```diff
@@ -13,37 +13,121 @@ Text editor operation to wrap the selection with the specified prefix and suffix
 "use strict";
 
 exports["wrap-selection"] = function(event,operation) {
-	if(operation.selStart === operation.selEnd) {
-		// No selection; check if we're within the prefix/suffix
-		if(operation.text.substring(operation.selStart - event.paramObject.prefix.length,operation.selStart + event.paramObject.suffix.length) === event.paramObject.prefix + event.paramObject.suffix) {
+	var o = operation,
+		prefix = event.paramObject.prefix,
+		prefixLength = prefix.length,
+		suffix = event.paramObject.suffix,
+		suffixLength = suffix.length,
+		trimSelection = event.paramObject.trimSelection || "no",
+		selLength = o.selEnd - o.selStart;
+
+	var trailingSpaceAt = function(sel) {
+		// returns "yes", "start", "end", "no"
+		var _start,
+			_end,
+			result;
+		// this evaluation takes the user configuration into account!
+		switch (trimSelection) {
+			case "end":
+				result = (sel.trimEnd().length !== selLength) ? "end" : "no";
+			break;
+			case "yes":
+				_start = sel.trimStart().length !== selLength;
+				_end = sel.trimEnd().length !== selLength;
+				result = (_start && _end) ? "yes" : (_start) ? "start" : (_end) ? "end" : "no";
+			break;
+			case "start":
+				result = (sel.trimStart().length !== selLength) ? "start" : "no";
+			break;
+			default:
+				result = "no";
+			break;
+		}
+		return result;
+	}
+
+	function togglePrefixSuffix() {
+		// this was the only behavour till TW v5.3.3
```

Comment:
I am not sure that comments like this are helpful. Most people looking at the code will have access to Git history if they have questions about the past. It is better to optimise the code readability experience for new people.